src/libANGLE/renderer/d3d/d3d11/renderer11_utils.cpp

```diff
@@ -2345,6 +2345,15 @@ HRESULT SetDebugName(ID3D11DeviceChild *resource,
                      const char *internalName,
                      const std::string *khrDebugName)
 {
+    UINT debugNameSize = 0;
+    HRESULT result = resource->GetPrivateData(WKPDID_D3DDebugObjectName, &debugNameSize, nullptr);
+
+    // If the resource already has a name, don't overwrite it.
+    if (debugNameSize > 0)
```

Comment:
```suggestion
    if (debugNameSize > 0 || result != S_OK)
```

Reply:
Hmm. I think both the original code and you suggestion make equal sense. But I'd rather use `!SUCCEEDED(result)` because probably we can try calling `SetPrivateData`, if `GetPrivateData` returned S_FALSE (note: `SUCCEEDED(S_FALSE)` is true)

Reply:
Eary returning on some unexpected `HRESULT` has to be slightly different, because `GetPrivateData()` returns [`DXGI_ERROR_NOT_FOUND`](https://learn.microsoft.com/en-us/windows/win32/direct3ddxgi/dxgi-error) in case the resource lacks an existing name. Implemented in https://github.com/shapr3d/angle/pull/21/commits/33298eb378acff596124047aba5894d8d6a3bd6d.